### PR TITLE
Add an enum file for node states

### PR DESCRIPTION
**Pull request by emurray2**
This may help with state management in the future (i.e. only having one state for being .playing or .paused).

In the past, there seems to be scenarios where the player has two states at a time (#2646, #2576, #2539). For example, maybe it is set to `isPlaying == true` in `play()`, but then set to `isPaused == true` when running `seek()`. I know that's probably not the case. However, maybe some states of the player would be easier to manage in an enum. Not every state would, but the significant ones might. A node probably shouldn't be playing and paused at the same time, but it might be playing and seeking.

Also, this article was sort of an inspiration behind this idea: https://dev.to/mpocock1/state-management-how-to-tell-a-bad-boolean-from-a-good-boolean-260n . I know web development is different, but this thinking may be useful here. Please let me know if anyone else has any ideas, feedback or thoughts on this. Thanks :) !

### CODE REVIEW

**Comment by aure** [comment]
DId you mean to include these tests in this PR?

**Comment by aure** [comment]
I think the enum is a good idea, but I don't think it should be committed until it is used and tested.

File: Tests/AudioKitTests/Node Tests/Player Tests/AudioPlayerTests.swift [hunk 1]
```diff
@@ -2,6 +2,66 @@ import AudioKit
...
+
+    let player = AudioPlayer()
+
+    init() {
+        mixer = Mixer()
+        mixer.addInput(player)
+        engine.output = mixer
+
+        do {
+            try Settings.setSession(category: .playAndRecord,
```
Comment (emurray2) [inline Tests/AudioKitTests/Node Tests/Player Tests/AudioPlayerTests.swift:19]: Weird, this compiles in Xcode. Not sure why it doesn't compile here. Could possibly be related to this? https://stackoverflow.com/questions/67494326/swiftui-cannot-infer-contextual-base-in-reference-to-member

**Comment by emurray2** [comment]
> DId you mean to include these tests in this PR?

I did, they seemed related to the same issue, but I think I'll open a separate one.

Sounds good, I'll work on testing them. Is there any way to get `AudioPlayer` to play sound during the tests? I can hear the oscillators, but I don't hear a sound when the player plays, so I don't know how to verify if it's working or not.

**Pull request closed without merging**